src/NServiceBus.AcceptanceTesting/Support/FailTestOnErrorMessageFeature.cs

```diff
@@ -12,15 +13,31 @@ public FailTestOnErrorMessageFeature()
             EnableByDefault();
 
             DependsOn<UnicastBus>();
+
+            RegisterStartupTask<FailTestOnErrorMessageFeatureStartupTask>();
         }
 
         protected internal override void Setup(FeatureConfigurationContext context)
         {
-            var endpointInstanceName = context.Settings.EndpointName();
-            var notifications = context.Builder.Build<BusNotifications>();
-            var testContext = context.Builder.Build<ScenarioContext>();
+        }
 
-            notifications.Errors.MessageSentToErrorQueue.Subscribe(new FaultedMessageObserver(testContext, endpointInstanceName));
+        class FailTestOnErrorMessageFeatureStartupTask : FeatureStartupTask
+        {
+            readonly BusNotifications notifications;
```

Comment:
And I know @SimonCropp will say :) that the `readonly` is redundant. Is there a way to have R# stop adding them?